### ISSUE TITLE
 The meger merge code will report an error 500 error

### ISSUE THREAD

**Issue by xingxing122** [gogs/gogs#6362]
cat gogs.log
2020/09/30 01:38:15 [ERROR] [...g/context/context.go:191 ServerError()] Merge: git merge --no-ff --no-commit [data/tmp/repos/284636393.git]: exit status 1 - 
2020/09/30 01:40:33 [ERROR] [...g/context/context.go:191 ServerError()] Merge: git merge --no-ff --no-commit [data/tmp/repos/910916864.git]: exit status 1 - 
2020/09/30 01:40:48 [ERROR] [...g/context/context.go:191 ServerError()] Merge: git merge --no-ff --no-commit [data/tmp/repos/780707494.git]: exit status 1 - 
2020/09/30 01:41:58 [ERROR] [...g/context/context.go:191 ServerError()] Merge: git merge --no-ff --no-commit [data/tmp/repos/576279750.git]: exit status 1 - 
2020/09/30 01:47:00 [ERROR] [...g/context/context.go:191 ServerError()] Merge: git merge --no-ff --no-commit [data/tmp/repos/256679763.git]: exit status 1 - 
2020/09/30 01:48:38 [ERROR] [...g/context/context.go:191 ServerError()] Merge: git merge --no-ff --no-commit [data/tmp/repos/674789665.git]: exit status 1 - 
2020/09/30 01:58:00 [ERROR] [...g/context/context.go:191 ServerError()] Merge: git merge --no-ff --no-commit [data/tmp/repos/419625341.git]: exit status 1 - 
2020/09/30 01:58:07 [ERROR] [...g/context/context.go:191 ServerError()] Merge: git merge --no-ff --no-commit [data/tmp/repos/210120483.git]: exit status 1 - 
2020/09/30 02:13:55 [ERROR] [...g/context/context.go:191 ServerError()] Merge: git merge --no-ff --no-commit [data/tmp/repos/248742914.git]: exit status 1 - 

gogs version:   gogs:0.11.91  
mysql version:  5.7  

app.ini  config
cat app.ini
APP_NAME = xxxxxxx
RUN_USER = git
RUN_MODE = prod

[database]
DB_TYPE  = mysql
HOST     = *******
NAME     = gogs
USER     = gogs
PASSWD   = ********
SSL_MODE = disable
PATH     = data/gogs.db

[repository]
ROOT = /data/git/gogs-repositories

[server]
DOMAIN       = ***********
HTTP_PORT    = 3000
ROOT_URL     = *********
DISABLE_SSH  = false
SSH_PORT     = 22
OFFLINE_MODE = false
[mailer]
ENABLED =  false
HOST= ******

[service]
REGISTER_EMAIL_CONFIRM = false
ENABLE_NOTIFY_MAIL     = false
DISABLE_REGISTRATION   = false
ENABLE_CAPTCHA         = true
REQUIRE_SIGNIN_VIEW    = false
ACTIVE_CODE_LIVE_MINUTES = 120
[picture]
DISABLE_GRAVATAR        = true
ENABLE_FEDERATED_AVATAR = false

[session]
PROVIDER = file

[log]
MODE      = file
LEVEL     = Info
ROOT_PATH = /data/app/gogs/log

[security]
INSTALL_LOCK = true
[webhook]
TYPES = gogs
SKIP_TLS_VERIFY = true

**Comment by unknwon** [comment]
Hi, please consider upgrade to 0.12.x and report back if the problem still exists :)

**Comment by jjakob** [comment]
I'm having the same error:
```
An error has occurred : git merge --no-ff --no-commit [/app/gogs/data/tmp/repos/573712022.git]: exit status 1 -

Application Version: 0.12.3
```
Using gogs/gogs:0.12 docker image.

**Comment by jjakob** [comment]
I think the problem was due to a merge conflict. The PR web UI said there are no merge conflicts, but when I tried to merge it on my local machine the merge failed with a conflict. There were some files committed in the topic branch under a submodule directory, where they should not have been (the author made a mistake and committed submodule changes into the main repo). So this is probably 2 issues: the PR web UI shows that there are no merge conflicts where there actually are (the merge fails) and the failed merge doesn't report an informative error message, but a general 500 error.
